x/mongo/driver/topology/sdam_spec_test.go

```diff
@@ -119,6 +119,13 @@ func setUpFSM(t *testing.T, uri string) *fsm {
 		fsm.Kind = description.Single
 	}
 
+	// GODRIVER-1578: The new test requires support for directConnection, which will be added in follow-up ticket
```

Comment:
@jyemin suggested doing both GODRIVER-1578 and GODRIVER-1486 together as one PR, but this PR is already on the longer side with minor refactoring/renames, so I'd prefer to keep them separate and have this temporary workaround here.

Reply:
Note for the future: it can be helpful for reviewers to separate the refactoring from the behavioral change.  They both can be done in the same PR, but if you do them in separate commits it is often much easier for reviewers to distinguish between the two types of changes rather than having to comb through a single commit and figure it out themselves.

Reply:
Ack

Reply:
Fair point. I'm not sure if it would've helped here because the first commit would've changed everything that created an `updateCallback` to set the right parameters and the second would've changed it to store the default update callback on the `Topology` instance itself, but good to keep in mind for future reviews.